en/releases/index.md

```diff
@@ -9,3 +11,107 @@ A list of PX4 release notes, they contain a list of the changes that went into e
 - [v1.12](../releases/1.12.md)
 
 The full archive of releases for the PX4 autopilot project can be found on [GitHub](https://github.com/PX4/PX4-Autopilot/releases)
+
```

Comment:
If the rest of the maintainers are happy with this, so am I.

One concern is that there are no time-frames for expectations. Now I know we can't say to a maintainer, you will have tested and fixed all issues by next week, but we can say something like  "the intent is that we'll converge within 2 months to a candidate in which only critical issues will be fixed, and 3 months for release from branch-off. Maintainers and tested are therefore expected to report on their progress to these goals" (or similar). Something that ties accountability.